File: Java/src/_0009_Palindrome_Number/Solution.java
package _0009_Palindrome_Number;

/**
 * Date: 2020/8/26 22:03
 * Content:
 */
public class Solution {

    public boolean isPalindrome(int x) {

        if (x < 0)
            return false;
        String s = x + "";

        int left = 0;
        int right = s.length() - 1;
        while (right > left) {
            if (s.charAt(left) != s.charAt(right))
                return false;
            else {
                left++;
                right--;
            }
        }
        return true;
    }

    public static void main(String[] args) {

        int x = 121;
        System.out.println((new Solution()).isPalindrome(x));
    }
}

File: Python/_0009_Palindrome_Number/Solution.py
# coding=utf-8


# 执行用时：76 ms, 在所有 Python3 提交中击败了58.76%的用户
# 内存消耗：14.8 MB, 在所有 Python3 提交中击败了80.07%的用户
class Solution:
    def isPalindrome(self, x: int) -> bool:
        return str(x) == str(x)[::-1]


if __name__ == "__main__":
    print(Solution().isPalindrome(121))  # True
    print(Solution().isPalindrome(-121))  # False
